src/app/main/component/shared/components/events-list-item/events-list-item.component.ts

```diff
@@ -279,6 +280,25 @@ export class EventsListItemComponent implements OnInit, OnDestroy {
     return this.langService.getLangValue(uaValue, enValue) as string;
   }
 
+  public flagHandler(): void {
+    this.isRegistered ? this.AddToEventPreviewList() : this.openAuthModalWindow('sign-in');
+  }
+
+  public AddToEventPreviewList(): void {
```

Comment:
this method doesn't make sense, you only created a local variable

Reply:
dded changes according to your comment